accepted/00061-hub-users.md

```diff
@@ -0,0 +1,128 @@
+
+Notes:
+- this will be "just a convenience feature" to let customers more easily define, inspect and apply security policies across many or all Servers. This RFC does not propose any real change of the model and mechanisms that currently exist
+- by leveraging Salt this solution benefits from a mechanism which is very flexible:
+  - SLS files allow to define a User/Org/SystemGroup/Permission structures as complicated as needed (from a single Server point of view), encouraging but not mandating uniformity across Servers
+  - [minion targeting](https://docs.saltstack.com/en/latest/topics/targeting/index.html) allows to select which Servers get which SLS states based on FQDNs/ids, grains, custom information (pillars), SystemGroups and so on
+- this proposal will not conflict with the use of PAM modules instead of the integrated authentication code
+- this proposal is expected to be orthogonal with any implementation of a Single Sign-On architecture
+
+## Relationship with the Hub gateway XMLRPC API
+
+The Hub _gateway_ XMLRPC API ([RFC here](accepted/00062-hub-xmlrpc-api.md)) gives access to clients from the Hub through Servers. Parts of that RFC that are directly impacted by this RFC are reproduced below.
+
+- Access to the gateway API itself requires Hub-User credentials. Access to individual Server APIs to manage clients requires Server-User credentials
+  - in worst case, N+1 pairs of credentials will be needed (one for the Hub, one per N Servers)
+  - as a special convenience option, the Hub gateway API can use Hub-User credentials to authenticate against Servers. So if a certain username/password was uniformly configured across several Servers (this is easy to implement with the states above), Hub gateway API usage becomes simpler: only one log in is needed, with its Hub-User credentials. After that a list of Servers to work with will have to be specified. This authentication mode can be optionally activated via a parameter (eg. `relayAuthentication = True`) on the `login` method
+  - as an additional special convenience option, the Hub gateway API can automatically select the list of Servers a Hub-User works with, based on the list of Servers that same Hub-User has permissions on. Once again this can be optionally activated via a parameter (eg. `autoConnect = True`)
+    - in this case only a single login with a pair of Hub-User credentials is needed, and then calls can be made directly
+    - Note that this only makes sense if the customer is OK with the idea of using Server sysadmin users to act on Clients
+    - If they prefer to manage Clients via users that have no permissions to manage Servers, they can create a Hub-Org with no Servers. Any Hub-User in this Hub-Org will be able to connect to the Hub gateway API, and possibly have permissions on Clients (if Servers have Server-Users with same name and passwords, again possibly configured via the above states). `autoConnect` won't be possible in those cases, at least initially
```

Comment:
If we use this solution how can users manage Software channels for example, if they will be part of a different organization of the clients?